### PR TITLE
account.update: set session ID from `x-set-session` response header

### LINKED ISSUE #75: account.update: set session ID from `x-set-session` response header
follow up for https://github.com/hoodiehq/hoodie-client-account/pull/74#discussion_r52942960

Right now, we send an internal `PUT /session` request after a `PATCH /session/account` request which changed the account username and/or password.

Instead, we want the `PATCH /session/account` response to optionally include a `x-set-session` header which would include the new session id, so that it can be updated without sending an extra request

---

**Working on your first Pull Request?** You can learn how from this *free* series [How to Contribute to an Open Source Project on GitHub](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) 

---

- [x] claim this issue (@gr2m)
- [x] [Setup repository locally](https://github.com/hoodiehq/hoodie-client-account#testing) and make sure that all tests pass
- [x] revert the changes of https://github.com/hoodiehq/hoodie-client-account/pull/74/files
- [x] update [test/integration/update-account-test.js#L19](https://github.com/hoodiehq/hoodie-client-account/blob/94844c3954c9d2bcc2e5d5f4e5127b01bec8f8ca/test/integration/update-account-test.js#L19): set a [reply header](https://www.npmjs.com/package/nock#specifying-reply-headers) for the `PATCH /session/account` response to include the `x-set-session` header with a new session id.
- [x] add [request headers](https://www.npmjs.com/package/nock#specifying-request-headers) to the `DELETE /session` request mock. Make sure the new session id is passed in the `Authorization: bearer <new session id here>` header. Make sure test fails when you run `npm test`
- [x] commit all test files with `test(update): session id`
- [x] fix the implementation in [lib/accounts/update.js](https://github.com/hoodiehq/hoodie-client-account/blob/94844c3954c9d2bcc2e5d5f4e5127b01bec8f8ca/admin/lib/accounts/update.js). We might need to change [utils/request.js](https://github.com/hoodiehq/hoodie-client-account/blob/94844c3954c9d2bcc2e5d5f4e5127b01bec8f8ca/utils/request.js) to resolve with both the response body & headers instead of just the body. Make sure tests pass when you run `npm test`
- [x] commit all changes with `fix(update): session id`
- [x] pull request: https://github.com/hoodiehq/hoodie-account-client/pull/94
- [x] Someone else: review & merge

Ping us in the [Hoodie Chat](http://hood.ie/chat/) or on [Twitter](https://twitter.com/hoodiehq/) if you have any questions :)

### PR DESCRIPTION
**Pull request by gr2m**
closes #75

### CODE REVIEW

**Comment by gr2m** [comment]
ping @hoodiehq/maintainers

File: lib/update.js [hunk 1]
```diff
@@ -25,21 +26,15 @@ function update (state, options) {
...
-      body: internals.serialise('session', options)
-    })
-  })
-
   .then(function (response) {
-    var data = internals.deserialise(response.body, {
-      include: 'account'
-    })
-    state.account.session.id = data.id
+    // when a username changes, the session ID gets recalculated, as it’s based
```
Comment (codyzu) [inline lib/update.js]: :+1:  for useful comments when they clarify the code

**Comment by codyzu** [comment]
LGTM nice work @gr2m !

**Comment by gr2m** [comment]
I’ll LGTM it myself because it’s blocking the Camp Release right now. But please review it anyway, maybe I missed something that we can fix with another PR

**Comment by codyzu** [comment]
I agree. Another review is a good idea... the modifications look sound to me, but I'm not intimate with the account-client (not yet) :-)

**Pull request merged**